partner_preview_scripts/ot_crispr.py

```diff
@@ -25,142 +34,166 @@ def __init__(self, study_url: str) -> None:
-        duplicated_study_ids = self.study_df.loc[lambda df: df.studyId.duplicated()].studyId.tolist()
-        assert len(duplicated_study_ids) == 0, f'Multiple studies have the same study id: {", ".join(duplicated_study_ids)}'
-
-        logging.info(f'Number of studies processed: {len(self.study_df.studyId.unique())}')
+        duplicated_study_ids = self.study_df.loc[
+            lambda df: df.studyId.duplicated()
+        ].studyId.tolist()
+        assert (
+            len(duplicated_study_ids) == 0
+        ), f'Multiple studies have the same study id: {", ".join(duplicated_study_ids)}'
+
+        logging.info(
+            f"Number of studies processed: {len(self.study_df.studyId.unique())}"
+        )
 
         projects = self.study_df.projectId.unique()
         logging.info(f'Number of projects: {len(projects)} ({", ".join(projects)})')
 
     def generate_evidence(self, data_folder: str) -> None:
-        """Processing the study dataframe"""
 
         # Looping through the studies and generating evidence:
         # Reading all data files and filter for significant hits:
         study_columns = [
-            'releaseDate',
-            'releaseVersion',
-            'studyId',
-            'dataFiles',
-            'dataFileType',
-            'filterColumn',
-            'threshold',
-            'projectId',
-            'ControlDataset',
-            'projectDescription'
+            "releaseDate",
+            "releaseVersion",
+            "studyId",
+            "dataFiles",
+            "dataFileType",
+            "filterColumn",
+            "threshold",
+            "projectId",
+            "ControlDataset",
+            "projectDescription",
         ]
 
         hits = (
             self.study_df[study_columns]
-            .explode('dataFiles')
+            .explode("dataFiles")
             .assign(
                 dataFile=lambda df: df.apply(
-                    lambda x: f'{data_folder}/{x["projectId"]}/{x["dataFiles"]}', axis=1)
+                    lambda x: f'{data_folder}/{x["projectId"]}/{x["dataFiles"]}', axis=1
+                )
             )
             .assign(
                 ControlDataset=lambda df: df.apply(
-                    lambda x: f'{data_folder}/{x["projectId"]}/{x["ControlDataset"]}' if pd.notna(x["ControlDataset"]) else None, axis=1)
+                    lambda x: f'{data_folder}/{x["projectId"]}/{x["ControlDataset"]}'
+                    if pd.notna(x["ControlDataset"])
+                    else None,
+                    axis=1,
+                )
             )
             # TODO: parsing the data files should be file type dependent:
             .apply(self.parse_MAGeCK_file, axis=1)
         )
 
         # Concatenate all hits into one single dataframe:
         hits_df = (
-            pd.concat(hits.to_list())
-            .reset_index(drop=True)
-
+            pd.concat(hits.to_list()).reset_index(drop=True)
```

Comment:
When I'm saying:

```python
pd.concat(hits.to_list())
```
- The values of a pd.Series is converted to a list.
- These values are pd.dataframes returned by the `.apply()` in line 90.
- These dataframes are concataneted into a single pd.DataFrame.

It might be a bit obscure. I'll update the variable name.